bpf/tc/attach.go

```diff
@@ -128,6 +128,24 @@ func (ap AttachPoint) AttachProgram() (string, error) {
 		}
 	}
 
+	// Check if the bpf object is already attached, and we should skip re-attaching it
+	hook := "tc_" + string(ap.Hook)
+	progID, err := ap.ProgramID()
+	if err != nil {
+		logCxt.WithError(err).Debug("Couldn't get the attached TC program ID. err=", err)
+	}
+
+	alreadyAttached, err := bpf.AlreadyAttachedProg(ap.IfaceName(), hook, preCompiledBinary, progID)
+	if err != nil {
+		logCxt.WithError(err).Debug("Failed to check if BPF program was already attached. err=", err)
+	}
```

Comment:
It should not be executed. The complete check needs 3 nested `if` statements, and it felt a hard to read and follow by nesting all those 3. So I arranged it as it is to make it easier to read. It executes some unnecessary instructions, but I felt it may not be a major issue since this code is only executed only during the start/restart of felix. What do you think?

Reply:
Ok, fixed it by using another method `AlreadyAttached()`